File: ruby/lib/steam/packets/rcon/rcon_auth_response.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'steam/packets/rcon/rcon_packet'

# This packet class represents a SERVERDATA_AUTH_RESPONSE packet sent by a
# Source server
#
# It is used to indicate the success or failure of an authentication attempt
# of a client for RCON communication.
#
# @author Sebastian Staudt
# @see SourceServer#rcon_auth
class RCONAuthResponse

  include RCONPacket

  # Creates a RCON authentication response for the given request ID
  #
  # The request ID of the packet will match the client's request if
  # authentication was successful
  #
  # @param [Fixnum] request_id The request ID of the RCON connection
  def initialize(request_id)
    super request_id, SERVERDATA_AUTH_RESPONSE, ''
  end

end

File: php/lib/steam/packets/rcon/RCONAuthResponse.php
<?php
/**
 * @author Sebastian Staudt
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package Steam Condenser (PHP)
 * @subpackage RCONAuthResponse
 * @version $Id$
 */

require_once "steam/packets/rcon/RCONPacket.php";

class RCONAuthResponse extends RCONPacket
{
  public function __construct($requestId)
  {
    parent::__construct($requestId, RCONPacket::SERVERDATA_AUTH_RESPONSE);
  }
}
?>
